From 69b11bda65a60182de803826937ea3b1c884df04 Mon Sep 17 00:00:00 2001
From: Martino Facchin <m.facchin@arduino.cc>
Date: Fri, 24 Apr 2015 12:27:02 +0200
Subject: [PATCH] allow cross compilation

---
 Makefile | 15 +++++++--------
 1 file changed, 7 insertions(+), 8 deletions(-)

diff --git a/Makefile b/Makefile
index 6133666..d258be4 100644
--- a/Makefile
+++ b/Makefile
@@ -63,7 +63,6 @@ endif
 #
 # Linux rules
 #
-ifeq ($(OS),Linux)
 COMMON_SRCS+=PosixSerialPort.cpp LinuxPortFactory.cpp
 COMMON_LIBS=-Wl,--as-needed
 WX_LIBS+=-lX11
@@ -72,7 +71,6 @@ MACHINE:=$(shell uname -m)
 
 install: strip
 	tar cvzf $(BINDIR)/bossa-$(MACHINE)-$(VERSION).tgz -C $(BINDIR) bossa$(EXE) bossac$(EXE) bossash$(EXE)
-endif
 
 #
 # OS X rules
@@ -134,15 +132,15 @@ DEPENDS+=$(BOSSASH_SRCS:%.cpp=$(OBJDIR)/%.d)
 # Tools
 #
 #Q=@
-CXX=g++
 ARM=arm-none-eabi-
+CXX=$(CROSS_COMPILE)g++
 ARMAS=$(ARM)as
 ARMOBJCOPY=$(ARM)objcopy
-
+STRIP=$(CROSS_COMPILE)strip
 #
 # CXX Flags
 #
-COMMON_CXXFLAGS+=-Wall -Werror -MT $@ -MD -MP -MF $(@:%.o=%.d) -DVERSION=\"$(VERSION)\" -g -O2
+COMMON_CXXFLAGS+= -MT $@ -MD -MP -MF $(@:%.o=%.d) -DVERSION=\"$(VERSION)\" -g -O2
 WX_CXXFLAGS:=$(shell wx-config --cxxflags --version=$(WXVERSION)) -DWX_PRECOMP -Wno-ctor-dtor-privacy -O2 -fno-strict-aliasing
 BOSSA_CXXFLAGS=$(COMMON_CXXFLAGS) $(WX_CXXFLAGS)
 BOSSAC_CXXFLAGS=$(COMMON_CXXFLAGS)
@@ -151,6 +149,7 @@ BOSSASH_CXXFLAGS=$(COMMON_CXXFLAGS)
 #
 # LD Flags
 #
+LD=$(CROSS_COMPILE)ld
 COMMON_LDFLAGS+=-g
 BOSSA_LDFLAGS=$(COMMON_LDFLAGS)
 BOSSAC_LDFLAGS=$(COMMON_LDFLAGS)
@@ -273,15 +272,15 @@ $(BINDIR)/bossash$(EXE): $(BOSSASH_OBJS) | $(BINDIR)
 
 strip-bossa: $(BINDIR)/bossa$(EXE)
 	@echo STRIP $^
-	$(Q)strip $^
+	$(Q)$(STRIP) $^
 
 strip-bossac: $(BINDIR)/bossac$(EXE)
 	@echo STRIP $^
-	$(Q)strip $^
+	$(Q)$(STRIP) $^
 
 strip-bossash: $(BINDIR)/bossash$(EXE)
 	@echo STRIP $^
-	$(Q)strip $^
+	$(Q)$(STRIP) $^
 
 strip: strip-bossa strip-bossac strip-bossash
 
-- 
2.3.1

