From 5a028bdf896d2dddb7a9cd6cf16c6a6fa165b094 Mon Sep 17 00:00:00 2001
From: Martino Facchin <m.facchin@arduino.cc>
Date: Fri, 24 Apr 2015 14:27:46 +0200
Subject: [PATCH 2/2] linux build as default

---
 Makefile | 2 --
 1 file changed, 2 deletions(-)

diff --git a/Makefile b/Makefile
index b0c1901..ff33beb 100644
--- a/Makefile
+++ b/Makefile
@@ -63,7 +63,6 @@ endif
 #
 # Linux rules
 #
-ifeq ($(OS),Linux)
 COMMON_SRCS+=PosixSerialPort.cpp LinuxPortFactory.cpp
 COMMON_LIBS=-Wl,--as-needed
 WX_LIBS+=-lX11
@@ -72,7 +71,6 @@ MACHINE:=$(shell uname -m)
 
 install: strip
 	tar cvzf $(BINDIR)/bossa-$(MACHINE)-$(VERSION).tgz -C $(BINDIR) bossa$(EXE) bossac$(EXE) bossash$(EXE)
-endif
 
 #
 # OS X rules
-- 
2.3.1

